modules/contracts/src.ts/services/ethService.ts

```diff
@@ -195,21 +208,29 @@ export class EthereumChainService extends EthereumChainReader implements IVector
     txFn: (gasPrice: BigNumber, nonce: number) => Promise<TransactionResponse | undefined>,
     gasPrice: BigNumber,
     signer: Signer,
+    chainId: number,
     nonce?: number,
   ): Promise<Result<TransactionResponse | undefined, Error>> {
     // Queue up the execution of the transaction.
-    return await this.queue.add(
-      async (): Promise<Result<TransactionResponse | undefined, Error>> => {
-        try {
-          // Send transaction using the passed in callback.
-          const actualNonce: number = nonce ?? (await signer.getTransactionCount("pending"));
-          const response: TransactionResponse | undefined = await txFn(gasPrice, actualNonce);
-          return Result.ok(response);
-        } catch (e) {
-          return Result.fail(e);
-        }
-      },
-    );
+    if (!this.queues.has(chainId)) {
+      return Result.fail(new ChainError(ChainError.reasons.SignerNotFound));
+    }
+    // Define task to send tx with proper nonce
+    const task = async (): Promise<Result<TransactionResponse | undefined, Error>> => {
+      try {
+        // Send transaction using the passed in callback.
+        const nonceToUse: number = nonce ?? this.nonces.get(chainId) ?? (await signer.getTransactionCount("pending"));
+        const response: TransactionResponse | undefined = await txFn(gasPrice, nonceToUse);
+        // After calling tx fn, set nonce to usedNonce + 1 IFF the
+        // nonce wasnt pre-specified (i.e. not re-sending tx)
+        await this.updateNonce(chainId, nonceToUse, signer);
```

Comment:
here we pass in `nonceToUse`, but the argument is labeled `usedNonce`

maybe to be more loyal to that, we could actually use `response.nonce` here?

Reply:
like, pass in the nonce that was actually used by the tx. I would imagine they would be the same, but this code likes to act UP

Reply:
It's specified to the `txFn` so I don't really think that matters, but sure